src/kernel/src/machine/pc/pcie.rs

```diff
@@ -1,26 +1,196 @@
+use alloc::collections::BTreeMap;
```

Comment:
I can't comment on a path/file so I'm commenting here.  Why machine/pc ?  I was wondering if this was an old holdover, but I don't see this in FreeBSD, is this a Linuxism or?

Reply:
This is an old holdover. I don't know what changes about PCIe as we move between platforms, so I put it here for now. We should move it to a more generic location, I agree.